packages/chevrotain/src/parse/errors_public.ts

```diff
@@ -82,7 +82,7 @@ export const defaultParserErrorProvider: IParserErrorMessageProvider = {
   }): string {
     const errPrefix = "Expecting: "
     // TODO: issue: No Viable Alternative Error may have incomplete details. #502
-    const actualText = first(actual).image
+    const actualText = first(actual)!.image
```

Comment:
Do we need this non-null assertion here because there is no type information?
or because we may be dealing with an empty array?
Apparently its possible to define a none empty array type
- https://stackoverflow.com/questions/56006111/is-it-possible-to-define-a-non-empty-array-type-in-typescript

Reply:
lodash's first always returns T | undefined. Changing actual to a non-empty array wouldn't change that. You could just do actual[0] instead, that wouldn't require a type assertion.

Reply:
I am starting to get the hang of using `!` for none empty arrays 👍